dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/InternalLogger.kt

```diff
@@ -105,9 +105,15 @@ interface InternalLogger {
      * [com.datadog.android.telemetry.model.TelemetryDebugEvent.Telemetry] event.
      * @param messageBuilder the lambda building the metric message
      * @param additionalProperties additional properties to add to the metric
+     * @param samplingRate value between 0-100 for sampling the event. Note that the sampling rate applied to this
+     * metric will be applied in addition to the global telemetry sampling rate.
      */
     @InternalApi
-    fun logMetric(messageBuilder: () -> String, additionalProperties: Map<String, Any?>)
+    fun logMetric(
+        messageBuilder: () -> String,
+        additionalProperties: Map<String, Any?>,
+        samplingRate: Float
```

Comment:
Will it be a good idea to give a default (also recommended) value for the sample rate? so in the place we cannot decide which sample rate we should use, we can have at least a good fallback.

Reply:
IMO, adding a default might lead into us not actively chosing a sampling rate. 
I'm all for providing sensible defaults on the public APIs used by our customers, but for our own internal code, I prefer forcing us to think about every param we use.

Reply:
Make sense to me